plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiSnapshotPageSource.java

```diff
@@ -0,0 +1,224 @@
+                    // Merge records from the page with log records
+                    for (int pos = 0; pos < page.getPositionCount(); pos++) {
+                        String recordKey = (String) avroSerializer.getValue(page, recordKeyFieldPos, pos);
+                        HoodieRecord logRecord = logRecordMap.remove(recordKey);
+                        if (logRecord != null) {
+                            // Merging base and log
+                            IndexedRecord baseRecord = avroSerializer.serialize(page, pos);
+                            Option<HoodieAvroIndexedRecord> mergedRecord = mergeRecord(baseRecord, logRecord);
+                            if (mergedRecord.isEmpty()) {
+                                continue;
+                            }
+                            avroSerializer.buildRecordInPage(pageBuilder, mergedRecord.get().getData(), partitionValueMap, true);
+                        } else {
+                            avroSerializer.buildRecordInPage(pageBuilder, page, pos, partitionValueMap, true);
+                        }
+                    }
+
+                    Page newPage = pageBuilder.build();
+                    pageBuilder.reset();
+                    return newPage;
+                } catch (IOException e) {
+                    throw new HoodieIOException("Cannot merge record in split " + split);
+                }
+            }
+        }
+
+        if (logRecordMap.isEmpty()) {
+            return null;
+        }
+
+        // Sending the rest to a page
+        for (HoodieRecord hudiRecord : logRecordMap.values()) {
+            IndexedRecord record = ((HoodieAvroIndexedRecord) hudiRecord).getData();
+            avroSerializer.buildRecordInPage(pageBuilder, record, partitionValueMap, true);
+        }
+
+        logRecordMap.clear();
+        Page newPage = pageBuilder.build();
+        pageBuilder.reset();
+        return newPage;
+    }
+
+    @Override
+    public long getMemoryUsage() {
+        return 0;
+    }
+
+    @Override
+    public void close() throws IOException {
+
+    }
+
+    private static HoodieMergedLogRecordScanner getMergedLogRecordScanner(HoodieStorage storage,
+                                                                          String basePath,
+                                                                          HudiSplit split,
+                                                                          Schema readerSchema) throws IOException {
+        return HoodieMergedLogRecordScanner.newBuilder()
+                .withStorage(storage)
+                .withBasePath(basePath)
+                .withLogFilePaths(split.logFiles())
+                .withReaderSchema(readerSchema)
+                .withLatestInstantTime(split.commitTime())
+                .withMaxMemorySizeInBytes(1 * 1024 * 1024L)
+                .withReverseReader(false)
+                .withBufferSize(1024 * 1024)
+                .withSpillableMapBasePath("/tmp/")
```

Comment:
let's move these hardcoded values to static consts.. In future (when we get to integrating with filegroup reader), we should evaluate if we need to make it configurable or not.

Reply:
Fixed.